data/scripts.js

```diff
@@ -1170,7 +1170,9 @@ let BattleScripts = {
 		// {gigantamax?: string, maxMoves: {[k: string]: string} | null}[]
 		if (!skipChecks) {
 			if (!pokemon.canDynamax) return;
-			if (pokemon.transformed && (pokemon.template.baseSpecies === 'zacian' || pokemon.template.baseSpecies === 'zamazenta' || pokemon.template.baseSpecies === 'eternatus')) return;
+			if (['zacian', 'zamazenta', 'eternatus'].includes(pokemon.template.baseSpecies) || pokemon.template.isMega || pokemon.template.isPrimal || pokemon.template.forme === "Ultra") {
+				return;
+			}
 			// Some pokemon species are unable to dynamax
 			const cannotDynamax = ['zacian', 'zamazenta', 'eternatus'];
 			if (cannotDynamax.includes(toID(pokemon.template.baseSpecies))) {
```

Comment:
Well in that case I'm missing how the three lines after your code don't duplicate the species checks...

Reply:
i have no clue, i'm about to look more into it